src/zino/tasks/juniperalarmtask.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from zino import state
+from zino.scheduler import get_scheduler
+from zino.snmp import SNMP
+from zino.statemodels import AlarmEvent, EventState
+from zino.tasks.task import Task
+
+_logger = logging.getLogger(__name__)
+
+
+class JuniperAlarmTask(Task):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._scheduler = get_scheduler()
+
+    async def run(self):
+        """Checks yellow and red alarm count for juniper device."""
+        device_state = state.devices.get(self.device.name)
+        if not device_state.is_juniper:
+            return
+
+        yellow_alarm_count, red_alarm_count = await self._get_juniper_alarms()
+
+        if type(yellow_alarm_count) is not int or type(red_alarm_count) is not int:
+            _logger.error(
+                "Device %s returns an alarm count that is not of type int, yellow alarm: %r, red alarm count: %r",
+                self.device.name,
+                yellow_alarm_count,
+                red_alarm_count,
+            )
+            return
+
+        device_state.alarms = {
+            "yellow": yellow_alarm_count,
+            "red": red_alarm_count,
+        }
+
+        yellow_alarm_event, yellow_created = state.events.get_or_create_event(
+            device_name=self.device.name,
+            port="yellow",
+            event_class=AlarmEvent,
+        )
+        if yellow_created:
+            yellow_alarm_event.state = EventState.OPEN
+            yellow_alarm_event.add_history("Change state to Open")
+        if yellow_alarm_event.alarm_count != yellow_alarm_count:
+            old_yellow_alarm_count = yellow_alarm_event.alarm_count
+            yellow_alarm_event.alarm_count = yellow_alarm_count
+            yellow_alarm_event.add_log(
+                f"{self.device.name} yellow alarms went from {old_yellow_alarm_count} to {yellow_alarm_count}"
+            )
+
+        red_alarm_event, red_created = state.events.get_or_create_event(
+            device_name=self.device.name,
+            port="red",
+            event_class=AlarmEvent,
+        )
+        if red_created:
+            red_alarm_event.state = EventState.OPEN
+            red_alarm_event.add_history("Change state to Open")
+        if red_alarm_event.alarm_count != red_alarm_count:
+            old_red_alarm_count = red_alarm_event.alarm_count
+            red_alarm_event.alarm_count = red_alarm_count
+            red_alarm_event.add_log(
+                f"{self.device.name} red alarms went from {old_red_alarm_count} to {red_alarm_count}"
+            )
```

Comment:
this function is pretty big, would prob be nice to split into one function handling red alarms and another handling yellow  or something along those lines